### PR TITLE
Support forcing a query to use the write pool

### PR DESCRIPTION
**Pull request by Willyham**
Resolves https://github.com/sequelize/sequelize/issues/3059

@sdepold @janmeier 

There were no tests for connection-manager that I could find, so I added some coverage for the code that I cleaned as well as added.

The diff is a little ugly because I inverted the big `if(config.replication)` condition :cry: 

As a bonus, I added the missing packages for generating the docs to package.json.

### CODE REVIEW

File: lib/sequelize.js [hunk 1]
```diff
@@ -652,6 +652,7 @@ module.exports = (function() {
    * @param {Boolean}         [options.nest=false] If true, transforms objects with `.` separated property names into nested objects using [dottie.js](https://github.com/mickhansen/dottie.js). For example { 'user.username': 'john' } becomes { user: { username: 'john' }}. When `nest` is true, the query type is assumed to be `'SELECT'`, unless otherwise specified
    * @param {Boolean}         [options.plain=false] Sets the query type to `SELECT` and return a single row
    * @param {Object|Array}    [options.replacements] Either an object of named parameter replacements in the format `:param` or an array of unnamed replacements to replace `?` in your SQL.
+   * @param {Boolean}         [options.forceWrite] Force the query to use the write pool, regardless of the query type.
```
Comment (janmeier) [inline lib/sequelize.js]: Remember to add the default arg (=false)

File: lib/dialects/abstract/connection-manager.js [hunk 2]
```diff
@@ -68,123 +67,123 @@ ConnectionManager.prototype.initPools = function () {
...
-    config.replication.read = config.replication.read.map(function (read) {
-      return _.extend({}, read);
-    });
+  if (!Array.isArray(config.replication.read)) {
+    config.replication.read = [config.replication.read];
+  }
+
+  // Make sure we don't modify the existing config object (user might re-use it)
+  config.replication.write = _.clone(config.replication.write);
+  config.replication.read = config.replication.read.map(function (read) {
```
Comment (janmeier) [inline lib/dialects/abstract/connection-manager.js]: We might be able to replace this with [cloneDeep](https://lodash.com/docs#cloneDeep)

Comment (janmeier) [inline lib/dialects/abstract/connection-manager.js]: (I know this was also in the original code, but I'm giving it all a review while I'm at it :)

**Comment by janmeier** [comment]
Looks solid. Just a couple of tiny comments and this should be good to go :)

Next time you don't need to commit the regenerated docs though - it makes the diff kinda hard to get an overview of. But thanks for adding the deps. to the package

File: lib/dialects/abstract/connection-manager.js (hunk 2 above)
Comment (Willyham) [inline lib/dialects/abstract/connection-manager.js]: Absolutely, I usually work with underscore, didn't realize lodash had a deep clone

**Comment by Willyham** [comment]
What do you think about the name of the option? I think `forceWrite` might be a little confusing if it's being used with a `.find`. Maybe `useWritePool` or `forceWritePool` or `useMaster`. Any opinons?

**Comment by janmeier** [comment]
`useMaster` sound good to me

**Comment by Willyham** [comment]
Rebase on upstream/master

**Comment by Willyham** [comment]
All tests on node0.10 pass, looks like 0.12/iojs is failing for an unrelated reason. Can you let me know if I need to do anything else here?

**Comment by janmeier** [comment]
There's a problem with your rebase in package.json. The mariasql dep has been removed, and sqlite updated to 3.0.5 on master. It think it might be easiest to take the dep list from master and just add the two docs-generation packages manually

**Comment by Willyham** [comment]
Pulled the latest package json and rebased. Will open a separate PR for the missing packages because this is starting to get painful!

**Comment by janmeier** [comment]
Thanks for you work @Willyham

**Pull request merged**